lib/src/main/kotlin/tsl/trees/AbstractBinaryTree.kt

```diff
@@ -0,0 +1,52 @@
+package tsl.trees
+
+import tsl.iterator.BinaryTreeIterator
+import tsl.nodes.AbstractNode
+
+abstract class AbstractBinaryTree<K : Comparable<K>, V, N : AbstractNode<K, V, N>> :
+    Iterable<Pair<K, V>> {
+    internal var root: N? = null
+
+    public abstract fun delete(key: K): V?
```

Comment:
```suggestion
    abstract fun delete(key: K): V?
```
непонятно зачем тут public. мы же не на жава пишем

Reply:
Иногда полезно определить видимость для людей, незнакомых с котлином. Зависит от аудитории